### ISSUE TITLE
Adding fetch_multi to Cache Strategy

### ISSUE THREAD

**Issue by joaomdmoura** [rails-api/active_model_serializers#827]
As pointed out by @thibaudgg on my last PR #810 (Adding Fragment Cache to AMS), [fetch_multi](http://api.rubyonrails.org/classes/ActiveSupport/Cache/Store.html#method-i-fetch_multi) would be a great optimisation to the new cache implementation (#693).

One feature of the new implementation is the individual cache strategy that enables your application to re-use the cache in different responses.

Ex. Use an object cached from a ```index``` method response on a ```show``` method response. The opposite is also true.

When retrieving this cached objects AMS will also do this individually, and that's when ```fetch_multi``` comes in. It improve performance retrieving multiple cache keys for a collection in a single go.

Despite of a simple improvement it isn't easy, this will change how the ```Adapter``` fetches cache and integrate with ```FragmentCache```   

I'm planning to work on this, but would like to hear some thoughts :)

**Comment by thibaudgg** [comment]
Great news, thanks @joaomdmoura!
I'm not familiar with the implementation, but IMHO adding `fetch_multi` seems really mandatory for production. 
It would works and reuse the object cached for associations as well right?

**Comment by joaomdmoura** [comment]
@thibaudgg I don't think it's mandatory (we are already conducting some production tests) but yeah, it it's definitively a great optimisation! Really useful! 
In this case it wouldn't be necessary in cached associations. The associations doesn't generate an individual cache for the associated object. It's cached along the original requested object.

**Comment by thibaudgg** [comment]
@joaomdmoura you're right "mandatory" is too strong, but definitely great to have.
My point with associations is to have each object cached separately, so for example if you load & cache a `post` with `comments` and you want to load just a `comment` after it should already be cached.
I guess that @kurko's [comment](https://github.com/rails-api/active_model_serializers/pull/810#issuecomment-77917040) goes in that direction.

**Comment by joaomdmoura** [comment]
Yeah @thibaudgg we discussed it yesterday and we'll move forward with this strategy! :smile: I'm not sure if ```fetch_multi``` will be used to associations yet, but they will be cache separately

**Comment by thibaudgg** [comment]
Great, keep up the good work!

**Comment by bf4** [comment]
The related thoughtbot article is pretty good, even though for 0.8 https://robots.thoughtbot.com/fast-json-apis-in-rails-with-key-based-caches-and

**Comment by philipgiuliani** [comment]
Already any updates on this? I noticed that most of the time my controller spends to talk with redis for fetching the records. Would be a great improvement!

**Comment by bf4** [comment]
Wanna help?

B mobile phone
